commands/http/handler.go

```diff
@@ -152,6 +153,9 @@ func (i internalHandler) ServeHTTP(w http.ResponseWriter, r *http.Request) {
 	if cn, ok := w.(http.CloseNotifier); ok {
 		go func() {
 			select {
+			case <-time.After(time.Minute * 30):
+				// TODO: this is a hack to avoid these goroutines building up in memory
+				log.Warning("TODO: close notify needs to be fixed")
```

Comment:
what if a file is legitimately big enough + over a low bw conn to require 30min to stream? again, this needs to take into account the rate + progress. a timeout like this is ok when there havent been any bytes through (or too few)